teps/0091-trusted-resources.md

```diff
@@ -394,6 +309,104 @@ What other approaches did you consider and why did you rule them out?  These do
+2. Avoid mutating Tekton Resources or move the mutating after verification.
+3. Store the encoded resource in annotation before mutating and verify the stored resource at validating webhook.
+4. Do verification in mutating webhook, if failed verification then mark the run as failed and later check and reject the resource in validationg webhook.
+
+**Pros vs Cons:**
+| Option | Pros | Cons |
+| -------- | ----------- | ----------- |
+| Call SetDefaults() when signing | Help users add missing values |  Need to update signature when there is update in mutating
+| Skip mutating webhook | Address the mutating issue, can also help to detect malicious mutating webhook| Extra cost for Tekton Pipeline developers
+| Store the object at mutating webhook before mutating| Fix the issue without skipping the mutating webhook |  May not work if the attck happens in mutating webhook and only modify the spec
+| Verify at mutating webhook | No need to skip the mutating or store in annotation | Doesn't work if there is a malicious mutating webhook after the normal mutating webhook
+
+In this TEP we proposed to proceed with option 1 and together with option 2.
+
+Option 1 can help to add missing values when users creating resources yaml files.
+
+A temporary solution for 2 is that we can have a specific feature flag for trusted resources and use it to skip the mutating.
+Options 2,3 and 4 are functionally equal,  i.e. make sure that when doing verification the resource is not mutated by mutating webhook.
+
+In current code base the mutating functions are called after we resolve the local/remote resources. Another benefit of option 2 is that we can prevent attack from malicious mutating webhooks. If there are  malicious mutating webhooks deployed in the cluster then the tampered resources will fail verification and be rejected at validating webhook.
+
+For option 3, it may be possible that a malicious mutating webhook is invoked after the tekton's mutating webhook and it can change the resource spec without touching the annotation. So to make it work we also need to compare the stored resource and current resource and make sure there is no malicious mutating happened.
+
+Option 4 is less preferred to 3 because mutating webhook needs to contain verification logic, and this is not well aligned with the intention of mutating webhook.
+
+### Alternatives for storing the signature
+
+In this TEP we proposed to store the signature in the `annotation` of the resource because of the easy implementation. Alternatives also include storing signature separately in alongside with the resource. This could be the future work of this TEP.
+
+### Alternatives for storing the public keys
+
+Option 1: Configure the keys in a dedicated CRD. This can help to support multiple keys in a central place.
```

Comment:
This is an interesting idea! Can you explain more clearly why it was rejected?

In particular: having keys installed in a separate CRD enables separation of authority -- security folks can centrally control which keys are valid for use and `task` / `pipeline` authors incorporate references to said keys. Now when a key needs to be rotated, the rotation happens in a single place (the CRD) rather than in every `task` / `pipeline` that is configured to use the kay.

Reply:
Oh this is not rejected, this is the option we propose in this TEP. But I list all options in the Alternatives section for better comparison. This is a idea suggested by Billy and inspired by  https://github.com/sigstore/policy-controller. 

I will add more details as you suggest in this PR! Thanks!